syntax = "proto3";

package salvo;

import "api/image.proto";
import "api/source.proto";
import "api/env.proto";

// This message type defines the schema for the consumed data file
// controlling the benchmark being executed. In it a user will
// define whether the benchmark uses images only, builds images
// from source, and whether the benchark executes locally or remotely.
message JobControl {
  // Specify whether the benchmark runs locally or in a service
  bool remote = 1;

  // Specify the benchmark to execute
  oneof benchmark {
    bool scavenging_benchmark = 2;
    bool dockerized_benchmark = 3;
    bool binary_benchmark = 4;
  }

  // Define where we find all required sources
  repeated SourceRepository source = 6;

  // Define the names of all required docker images
  DockerImages images = 7;

  // Define the environment variables needed for the test
  EnvironmentVars environment = 8;
}
